tests/components/blebox/test_switch.py

```diff
@@ -0,0 +1,386 @@
+    feature = mock_feature(
+        "switches",
+        blebox_uniapi.switch.Switch,
+        unique_id="BleBox-switchBox-1afe34e750b8-0.relay",
+        full_name="switchBox-0.relay",
+        device_class="relay",
+        is_on=False,
+    )
+    feature.async_update = AsyncMock()
+    product = feature.product
+    type(product).name = PropertyMock(return_value="My switch box")
+    type(product).model = PropertyMock(return_value="switchBox")
+    return (feature, "switch.switchbox_0_relay")
+
+
+async def test_switchbox_init(switchbox, hass, config):
+    """Test switch default state."""
+
+    feature_mock, entity_id = switchbox
+
+    feature_mock.async_update = AsyncMock()
+    entry = await async_setup_entity(hass, config, entity_id)
+    assert entry.unique_id == "BleBox-switchBox-1afe34e750b8-0.relay"
+
+    state = hass.states.get(entity_id)
+    assert state.name == "switchBox-0.relay"
+
+    assert state.attributes[ATTR_DEVICE_CLASS] == DEVICE_CLASS_SWITCH
+
+    assert state.state == STATE_OFF
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(entry.device_id)
+
+    assert device.name == "My switch box"
+    assert device.identifiers == {("blebox", "abcd0123ef5678")}
+    assert device.manufacturer == "BleBox"
+    assert device.model == "switchBox"
+    assert device.sw_version == "1.23"
+
+
+async def test_switchbox_update_when_off(switchbox, hass, config):
+    """Test switch updating when off."""
+
+    feature_mock, entity_id = switchbox
+
+    def initial_update():
+        feature_mock.is_on = False
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    await async_setup_entity(hass, config, entity_id)
+
+    state = hass.states.get(entity_id)
+    assert state.state == STATE_OFF
+
+
+async def test_switchbox_update_when_on(switchbox, hass, config):
+    """Test switch updating when on."""
+
+    feature_mock, entity_id = switchbox
+
+    def initial_update():
+        feature_mock.is_on = True
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    await async_setup_entity(hass, config, entity_id)
+
+    state = hass.states.get(entity_id)
+    assert state.state == STATE_ON
+
+
+async def test_switchbox_on(switchbox, hass, config):
+    """Test turning switch on."""
+
+    feature_mock, entity_id = switchbox
+
+    def initial_update():
+        feature_mock.is_on = False
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    await async_setup_entity(hass, config, entity_id)
+    feature_mock.async_update = AsyncMock()
+
+    def turn_on():
+        feature_mock.is_on = True
+
+    feature_mock.async_turn_on = AsyncMock(side_effect=turn_on)
+
+    await hass.services.async_call(
+        "switch", SERVICE_TURN_ON, {"entity_id": entity_id}, blocking=True,
+    )
+
+    state = hass.states.get(entity_id)
+    assert state.state == STATE_ON
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5535891)